core/controller/src/main/scala/whisk/core/controller/Meta.scala

```diff
@@ -134,14 +136,54 @@ protected[core] object WhiskMetaApi extends Directives {
                     case _ => throw new Throwable("Illegal code")
                 } getOrElse (OK)
 
-                respondWithHeaders(headers) {
-                    complete(code)
+                fields.get("body") map {
+                    case JsString(str) => interpretHttpResponse(code, headers, str, transid)
+                    case _             => terminate(BadRequest, Messages.httpContentTypeError)(transid)
+                } getOrElse {
+                    respondWithHeaders(headers) {
+                        // note that if header defined a content-type, it will be ignored
+                        // since the type must be compatible with the data response
+                        complete(code)
+                    }
                 }
+
             } getOrElse terminate(BadRequest, Messages.invalidMedia(`message/http`))(transid)
 
             case _ => terminate(BadRequest, Messages.invalidMedia(`message/http`))(transid)
         }
     }
+
+    private def interpretHttpResponse(code: StatusCode, headers: List[RawHeader], str: String, transid: TransactionId): RequestContext => Unit = {
+        Try {
+            // header must be one of known headers, else reject it;
+            // if not specified, assume text/html.
+            headers find {
+                _.lowercaseName == HttpHeaders.`Content-Type`.lowercaseName
+            } map { t =>
+                // per spec, main/sub type and parameters are separated by ';'
+                // if this fails, it's not a valid header
+                val parts = t.value.split(';')
+                val mainAndSubtype = parts(0).split('/')
+                val contentType = mainAndSubtype(0).toLowerCase -> mainAndSubtype(1).toLowerCase
```

Comment:
Might want to use destructuring, to get rid of the index getters:

```
val Array(type, _) = t.value.split(';')
val Array(mainType, subType) = type.split('/')
```

improves readability imho.